File: 0508-most-frequent-subtree-sum/0508-most-frequent-subtree-sum.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int getfreq(TreeNode node,HashMap<Integer,Integer> map) {
        if(node == null) return 0;
        
        int leftSum = getfreq(node.left,map);
        int rightSum = getfreq(node.right,map);
        
        int totalSum = leftSum+rightSum+node.val;
        map.put(totalSum,map.getOrDefault(totalSum,0)+1);
        
        return totalSum;
    }
    public int[] findFrequentTreeSum(TreeNode root) {
        HashMap<Integer,Integer> map = new HashMap<>();
        getfreq(root,map);
        
        int maxFreq = -1;
        int count=0;
        for(int value:map.values()) {
            if(maxFreq == value) count++;
            if(maxFreq < value) {
                count=1;
                maxFreq = value;
            }
        }
        System.out.print(count);
        int[] ans = new int[count];
        int i=0;
        for(int key:map.keySet()) {
            if(map.get(key) == maxFreq) ans[i++] = key;
        }
        return ans;
    }
}

File: 0508-most-frequent-subtree-sum/0508-most-frequent-subtree-sum.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int  getfreq(TreeNode* node,unordered_map<int,int>& map) {
        if(!node) return 0;
        
        int leftS = getfreq(node->left,map);
        int rightS = getfreq(node->right,map);
        
        int totalS = leftS + rightS + node->val;
        
        map[totalS]++;
        
        return totalS;
    }
    vector<int> findFrequentTreeSum(TreeNode* root) {
        unordered_map<int,int> map;
        vector<int> ans;
        getfreq(root,map);
        int maxFreq = -1;
        for(auto it :map) {
            if(it.second > maxFreq) maxFreq = it.second;
        }
        for(auto it:map) {
            if(it.second == maxFreq) ans.push_back(it.first);
        }
        return ans;
    }
};